tests/test_template.py

```diff
@@ -0,0 +1,52 @@
+import unittest
+import numpy as np
+from pytom_tm.template import generate_template_from_map
+
+
+class TestTemplate(unittest.TestCase):
+    def setUp(self):
+        self.template = np.zeros((13, 13, 13), dtype=np.float32)
+        self.template[3:5, 3:5, 7:9] = 1
+
+    def test_template_padding(self):
+        uneven_box = np.zeros((13, 13, 7))
+        new_template = generate_template_from_map(uneven_box, 1, 1)
+        self.assertEqual(
+            new_template.shape,
+            self.template.shape,
+            msg="Box should be made square"
+        )
+        new_template = generate_template_from_map(
+            uneven_box, 1, 2, output_box_size=20)
+        self.assertEqual(
+            new_template.shape,
+            (20, ) * 3,
+            msg="Template was not padded to output box size"
+        )
+
+    def test_template_centering(self):
+        new_template = generate_template_from_map(
+            self.template, 1, 1, center=False,
+        )
+        square_sum = np.square(new_template - self.template).sum()
+        self.assertTrue(square_sum < 1, msg="Template should not change strongly "
+                                            "without recentering.")
+        new_template = generate_template_from_map(
+            self.template, 1, 1, center=True,
+        )
+        square_sum = np.square(new_template - self.template).sum()
+        self.assertTrue(square_sum > 1, msg="Template didnt change after shift")
```

Comment:
Do you think it makes sense to make sure this test tests mixed high and low values in the template? (to catch #173)

Reply:
My idea was to add those things in for that PR and here just prep the test for that one.

Reply:
:+1: